plaidml/edsl/edsl_test.cc

```diff
@@ -46,6 +46,50 @@ Tensor Softmax(const Tensor& X) {
   return E / N;
 }
 
+TEST(CppEdsl, LowerPrecisionInvalidNegative) {
+  auto A = Placeholder(DType::FLOAT32, {3, 3});
+  auto C = A * (-2);
+
+  EXPECT_ANY_THROW({ Program("lower_precision", {C}, DType::FLOAT16, DType::UINT16); });
+}
+
+TEST(CppEdsl, LowerPrecision) {
```

Comment:
I'd call this `LowerPrecisionConstants` because it's only modifying the constant types, it's not causing the entire test to produce lower precision results.